libs/client/WebhookClient.lua

```diff
@@ -0,0 +1,314 @@
+		if #self._listeners[name] > 0 then
+			return self:emit(name, msg or format(fmt, ...))
+		end
+	end
+end
+
+-- Getters
+function get.avatarURL(self, size, ext)
+	return self._options.avatarURL ~= "" and self._options.avatarURL or resolveImage(self._options.avatar, self._id)
+end
+
+function get.channelId(self)
+	return self._options.channel_id
+end
+
+function get.guildId(self)
+	return self._webhook.guildId
+end
+
+function get.id(self)
+	return self._webhook.id
+end
+
+function get.name(self)
+	return self._options.name
+end
+
+function get.token(self)
+	return self._webhook.token
+end
+
+function get.user(self)
+	return clean(self._webhook.user) -- does not return a full user object because it is not linked with Client
+end
+
+
+-- Setters and Modifiers
+function WebhookClient:setAvatar(avatar)
+	self._options.avatarURL = avatar
+	avatar = resolveImage(avatar) or Resolver.base64(avatar) or null
+
+	local bool, err = self:modify({avatar = avatar})
+	if bool then
+		return self
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:setChannelId(channelId)
+	self._options.channelId = channelId
+	local bool, err = self:modify({channel_id = channelId})
+
+	if bool then
+		return self
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:setName(name)
+	self._options.name = name
+	local bool, err = self:modify({name = name})
+
+	if bool then
+		return self
+	else
+		self:error(err)
+	end
+end
+
+function WebhookClient:modify(tbl)
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
+		self._options[k] = v
+	end
+
+	return modify(self, tbl)
+end
+
+function WebhookClient:modifyWithToken(tbl) 
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
+		if k ~= "channelId" then -- You can not change the channelId
+			self._options[k] = v
+		else
+			self:info("modifyWithToken does not accept a channelId key. Use 'modify' to change it.")
+		end
+	end
+	self._api:modifyWebhookWithToken(self._id, self._token, tbl)
+
+	return self
+end
+
+-- Rest of functions
+function WebhookClient:send(content, options) -- return message
+	local file
+	if options then
+		for key, value in pairs(options) do
+			self._options[key] = value -- merge new options
+		end
+
+		if options.file then -- Only one file can be sent at the same time.
+			cntn, err = parseFile(options.file)
+			if err then
+				return nil, err
+			end
+			file = cntn
+		end
+	end
+
+	if file then
+		self._options.file = file
+	end
+
+	local webhookData = self._options
+	local data, err = self._api:executeWebhook(self._id, self._token, { -- We insert these values manually so that we don't send extra (irrevelant) information
+		content = content,
+		username = webhookData.name,
+		avatar_url = webhookData.avatarURL,
+		tts = webhookData.tts,
+		embeds = webhookData.embeds
+	}, webhookData.file)
+
+	if data then
+		return Message(data, self), self
```

Comment:
I don't know if this should return a Message or just true/false, or a custom WebhookMessage object. Is is possible to write a WebhookClient that does not rely on Discordia containers?

Reply:
We can return a boolean as of now and work on creating a custom WebhookMessage with the limited data Discord returns when waiting for the object creation.  It is possible to write a WebhookClient without relying on Discordia containers. In my rewrite I don't use any of them in the WebhookClient.